apps/arena/lib/arena/game/bounties.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Arena.Game.Bounties do
+  @moduledoc """
+  Module to handle quest logic in arena application
+  """
+
+  defp parse_comparator("equal"), do: &Kernel.==/2
+  defp parse_comparator("distinct"), do: &Kernel.!=/2
+  defp parse_comparator("greater"), do: &Kernel.>/2
+  defp parse_comparator("greater_or_equal"), do: &Kernel.>=/2
+  defp parse_comparator("lesser"), do: &Kernel.</2
+  defp parse_comparator("lesser_or_equal"), do: &Kernel.<=/2
+  defp parse_comparator(comparator), do: raise("Comparator not implemented yet #{comparator}")
+
+  defp accumulate_objective_progress_by_scope("day", value), do: value
+  defp accumulate_objective_progress_by_scope("match", _value), do: 1
+
+  def completed_bounty?(nil, _arena_match_results) do
+    false
+  end
+
+  def completed_bounty?(quest, arena_match_results) do
+    progress =
+      arena_match_results
+      |> filter_results_that_meet_quest_conditions(quest.conditions)
+      |> Enum.reduce(0, fn arena_match_result, acc ->
+        type = String.to_atom(quest.objective.match_tracking_field)
```

Comment:
This is HIGHLY DISCOURAGED.
I already mentioned something regarding Atoms in another comment. Remember atoms ARE NOT garbage collected, we must not create atoms in an uncontrolled way like this one.

Reply:
Thanks for clarifying this! I took the approach you mentioned to create a function to pattern match the field and turn it into an atom if we want to, wrong fields will be taken as invalid and won't be taken into account when checking if a quest is completed